### DISCUSSION TITLE
Add additional field to database when files are uploaded via media library

### DISCUSSION THREAD

**Discussion by usvs-steve** [strapi/strapi#8159] [Legacy Topics]
Hi all, sorry if this is an easy question but I'm a bit stuck.

I need to be able to add an additional field to the mongoDB document that is created when a file is uploaded to the Strapi media library. After looking through the Strapi application and specifically within the Node Modules I have found the folder named "strapi-plugin-upload" and within this have edited the Upload.js file with the following location - node_modules/strapi-plugin-upload/services/Upload.js 

I made amends to the file within the "formatFileInfo" function and added the extra field field called "partitionKey" within the const "entity".

    const entity = {
      name: usedName,
      alternativeText: fileInfo.alternativeText,
      caption: fileInfo.caption,
      hash: generateFileName(usedName),
      ext,
      mime: type,
      size: bytesToKbytes(size),
      partitionKey: "Lorem Ipsum"
    };

<img width="491" alt="Screenshot 2020-10-02 at 15 54 14" src="https://user-images.githubusercontent.com/7020952/94937505-92a19c80-04c7-11eb-855e-ee9735654950.png">

I then rebuilt the application using "npm run build" however the extra field does not appear in the mongodb database. I feel like this is the correct location to make amends so I'm obviously missing something. Any help would be greatly appreciated.

Thanks in advance.

**Comment by usvs-steve** [comment]
Ignore found the answer immediately after writing this question as is always the way.

For anyone that wants to know I actually needed to amend the "File.settings.json" file within the models folder.

Thanks all

**Reply by derrickmehaffy** [reply]
Yup that is usually the case :wink: glad you found it!
